1Password/docusaurus-extensions
# [DEV] Typescript 5.2.2 + support new Docusaurus tsconfig

## Summary

Docusaurus 3.0 requires `typescript` to 5.1+. This MR updates project deps to 5.2.2 and supports the new Docusaurus tsconfig via `@docusaurus/tsconfig`.

## Related Issues and MRs

Closing #53 

## What changed

- Upgrade `typescript` to 5.2.2 within the website workspace
- Remove `@tsconfig/docusaurus` and replace with `@docusaurus/tsconfig`
- Extend new tsconfig from `tsconfig.json`

## Steps for reviewer

1. Pull changes and install
```
git checkout brian/mdx-3
yarn install
```
2. Verify `typescipt` has been updated throughout the project
3. Verify our plugins build properly
```
yarn dev docusaurus-plugin-stored-data
```
```
yarn dev jest-docusaurus
```
4. Ensure project has been typechecked
```
yarn typecheck
```

## Additional resources

- https://docusaurus.io/docs/migration/v3#typescript-v51
